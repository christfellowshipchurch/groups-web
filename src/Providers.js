import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';

import { defaultTheme } from './ui-kit/theme';

const Providers = ({ children }) => (
  <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
);

Providers.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export default Providers;
