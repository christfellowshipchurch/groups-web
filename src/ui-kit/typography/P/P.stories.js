import React from 'react';
import { storiesOf } from '@storybook/react';

import P from '.';

storiesOf('ui-kit/typography/P', module)
  .add('default', () => (
    <P>
      {`But I, brothers, could not address you as spiritual people, but as people of the flesh, as infants in Christ. I fed you with milk, not solid food, for you were not ready for it. And even now you are not yet ready, for you are still of the flesh. For while there is jealousy and strife among you, are you not of the flesh and behaving only in a human way? For when one says, "I follow Paul," and another, "I follow Apollos," are you not being merely human?`}
    </P>
  ))
  .add('isLoading', () => (
    <P isLoading>
      {`But I, brothers, could not address you as spiritual people, but as people of the flesh, as infants in Christ. I fed you with milk, not solid food, for you were not ready for it. And even now you are not yet ready, for you are still of the flesh. For while there is jealousy and strife among you, are you not of the flesh and behaving only in a human way? For when one says, "I follow Paul," and another, "I follow Apollos," are you not being merely human?`}
    </P>
  ));
