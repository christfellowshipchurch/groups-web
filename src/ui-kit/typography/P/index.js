import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { Paragraph as Placeholder } from '../../Placeholder';
import { getIsLoading } from '../../isLoading';

const StyledP = styled.p`
  font-size: 1rem;
  line-height: ${({ theme }) => theme.helpers.verticalRhythm(1)};
  margin: 0 0 ${({ theme }) => theme.helpers.verticalRhythm(1)} 0;
  ${({ customStyle }) => customStyle}
`;

const P = getIsLoading(({ children, isLoading, style, ...props }) => (
  <Placeholder isLoading={isLoading} showLoadingAnimation>
    <StyledP customStyle={style} {...props}>
      {children}
    </StyledP>
  </Placeholder>
));

P.defaultProps = {
  isLoading: false,
};

P.propTypes = {
  children: PropTypes.node.isRequired,
  isLoading: PropTypes.bool,
  style: PropTypes.any, // eslint-disable-line
};

P.displayName = 'ui-kit.typography.P';

export default P;
