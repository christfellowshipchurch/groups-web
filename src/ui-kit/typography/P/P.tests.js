import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Providers from '../../../Providers';

import P from '.';

describe('the P component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <P>
          {`But I, brothers, could not address you as spiritual people, but as people of the flesh, as infants in Christ. I fed you with milk, not solid food, for you were not ready for it. And even now you are not yet ready, for you are still of the flesh. For while there is jealousy and strife among you, are you not of the flesh and behaving only in a human way? For when one says, "I follow Paul," and another, "I follow Apollos," are you not being merely human?`}
        </P>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const salmonText = { color: 'salmon' };
    const tree = renderer.create(
      <Providers>
        <P style={salmonText}>
          {`But I, brothers, could not address you as spiritual people, but as people of the flesh, as infants in Christ. I fed you with milk, not solid food, for you were not ready for it. And even now you are not yet ready, for you are still of the flesh. For while there is jealousy and strife among you, are you not of the flesh and behaving only in a human way? For when one says, "I follow Paul," and another, "I follow Apollos," are you not being merely human?`}
        </P>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render a loading state', () => {
    const tree = renderer.create(
      <Providers>
        <P isLoading>Loading state</P>
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
