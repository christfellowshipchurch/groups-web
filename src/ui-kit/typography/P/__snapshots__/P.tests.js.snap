// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the P component should accept and render passed in styles 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.5rem;
  margin: 0 0 1.5rem 0;
  color: salmon;
}

<p
  className="c0 sc-1ec6ydc-1 cmTtod"
>
  But I, brothers, could not address you as spiritual people, but as people of the flesh, as infants in Christ. I fed you with milk, not solid food, for you were not ready for it. And even now you are not yet ready, for you are still of the flesh. For while there is jealousy and strife among you, are you not of the flesh and behaving only in a human way? For when one says, "I follow Paul," and another, "I follow Apollos," are you not being merely human?
</p>
`;

exports[`the P component should render 1`] = `
.c0 {
  font-size: 1rem;
  line-height: 1.5rem;
  margin: 0 0 1.5rem 0;
}

<p
  className="c0 sc-1ec6ydc-1 cmTtod"
>
  But I, brothers, could not address you as spiritual people, but as people of the flesh, as infants in Christ. I fed you with milk, not solid food, for you were not ready for it. And even now you are not yet ready, for you are still of the flesh. For while there is jealousy and strife among you, are you not of the flesh and behaving only in a human way? For when one says, "I follow Paul," and another, "I follow Apollos," are you not being merely human?
</p>
`;

exports[`the P component should render a loading state 1`] = `
Array [
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 85%;
}

.c1 {
  height: 1rem;
  margin: calc(1.5rem - 1rem) 0;
}

<mark
    className="c0 show-loading-animation ra2rjz-1 c1 custom-placeholder"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: calc(1.5rem - 1rem) 0;
}

<mark
    className="c0 show-loading-animation ra2rjz-1 c1 custom-placeholder"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 75%;
}

.c1 {
  height: 1rem;
  margin: calc(1.5rem - 1rem) 0;
}

<mark
    className="c0 show-loading-animation ra2rjz-1 c1 custom-placeholder"
  />,
]
`;

exports[`the P component should render a loading state via context 1`] = `
Array [
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 85%;
}

.c1 {
  height: 1rem;
  margin: calc(1.5rem - 1rem) 0;
}

<mark
    className="c0 show-loading-animation ra2rjz-1 c1 custom-placeholder"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 100%;
}

.c1 {
  height: 1rem;
  margin: calc(1.5rem - 1rem) 0;
}

<mark
    className="c0 show-loading-animation ra2rjz-1 c1 custom-placeholder"
  />,
  .c0 {
  background: #A5A5A5;
  border-radius: 16px;
  display: block;
  height: 1rem;
  width: 75%;
}

.c1 {
  height: 1rem;
  margin: calc(1.5rem - 1rem) 0;
}

<mark
    className="c0 show-loading-animation ra2rjz-1 c1 custom-placeholder"
  />,
]
`;
